File: sdks/r/demo/quickstart.R
# nRouter — R hello world
# install.packages("nrouter", repos = c(nrouterai = "https://nrouterai.r-universe.dev", CRAN = "https://cloud.r-project.org"))

library(nrouter)

# A Smart Router alias activates its strategy/fallback chain; a concrete model
# id pins the request to that model.
model <- Sys.getenv("NROUTER_MODEL", "gpt-5.4-mini")
client <- nrouter_client()

response <- nrouter_chat_completions(client, list(
  model = model,
  messages = list(list(role = "user", content = "Hello, nRouter!"))
))

cat(response$body$choices[[1]]$message$content, "\n")
print(response$meta)

File: sdks/rust/demo/quickstart.rs
// nRouter — Rust hello world
// Cargo.toml:
//   nrouter = "2.1"
//   serde_json = "1"
//   tokio = { version = "1", features = ["full"] }

use nrouter::http::Client;
use serde_json::json;

#[tokio::main]
async fn main() {
    let client = Client::from_env().expect("NROUTER_API_KEY not set");
    // A Smart Router alias activates its strategy/fallback chain; a concrete
    // model id pins the request to that model.
    let model = std::env::var("NROUTER_MODEL")
        .unwrap_or_else(|_| "gpt-5.4-mini".to_string());
    let response = client.chat_completions(&json!({
        "model": model,
        "messages": [{"role": "user", "content": "Hello, nRouter!"}]
    })).await.unwrap();
    println!("{}", response.body["choices"][0]["message"]["content"]);
    println!("request {:?}, cost {:?}", response.meta.request_id, response.meta.cost);
}
